sample-notificator-app/src/main/groovy/life/qbic/samplenotificator/components/EmailGenerator.groovy

```diff
@@ -1,48 +1,65 @@
 package life.qbic.samplenotificator.components
 
-import life.qbic.business.subscription.Subscriber
+import life.qbic.business.notification.create.NotificationContent
+import org.jsoup.Jsoup
+import org.jsoup.nodes.Document
 
 import java.util.concurrent.TimeUnit
 
+
 /**
  * Class responsible for generating an sending an email from the provided notification and subscriber information
  *
  * EmailGenerator converts the notification created in the CreateNotification Use case into an emailBody
  * The contactInformation such as e.g. the emailAddress of the recipient is retrieved from the associated provided Subscriber
  * The email is then send via a commandline call to the mailutils tool(@link <a href=https://mailutils.org/>mailutils</a> via a ProcessBuilder
  *
- * @since: 1.0.0
- *
+ * @since: 1.0.0*
  */
 class EmailGenerator {
 
     private String subject = "Project Update"
-    Map<Subscriber, String> notificationPerSubscriber
+    List<NotificationContent> notificationContentList
+    private InputStream EMAIL_HTML_TEMPLATE_STREAM
+    private EmailHTMLTemplate emailHTMLTemplate
+    private List<Document> filledTemplates = []
 
-    EmailGenerator(Map<Subscriber, String> notificationPerSubscriber) {
-        this.notificationPerSubscriber = notificationPerSubscriber
+    EmailGenerator(List<NotificationContent> notificationContentList) {
+        this.notificationContentList = notificationContentList
     }
 
     /**
      * Triggers the creation and submission of each email
      * containing the provided notifications to the provided subscribers
      */
-    void initializeEmailSubmission(){
-        notificationPerSubscriber.each {
-            //ToDo Replace this with subscriber mail
-            sendEmail("Steffen.greiner@uni-tuebingen.de", it.value)
+    void initializeEmailSubmission() {
+        notificationContentList.each { NotificationContent notificationContent ->
+            accessEmailTemplate()
+            prepareHTMLEmail(notificationContent)
+            //ToDo sendEmail will be handled by a different PR
+            //sendEmail("Steffen.greiner@uni-tuebingen.de", filledTemplate.html())
         }
     }
 
+    private void accessEmailTemplate() {
+        this.EMAIL_HTML_TEMPLATE_STREAM = EmailHTMLTemplate.class.getClassLoader().getResourceAsStream("notification-template/email-update-template.html")
+    }
```

Comment:
why is that not a classvariable? this will be the same for this document, right?

Reply:
True! However, The InputStream is consumed every time it's parsed, therefore it has to be generated anew for each notification.

Reply:
Is it? I don't think so. You write the input into the emailHTMLTemplate variable. I just checked the OM. By making it a class variable you do not only save time but also unnecessary file reading. Furthermore, your method would look cleaner (the method where you call this). 

I even think its better to store the URI rather than a stream. In the OM we had
```
  private static final URI OFFER_HTML_TEMPLATE_URI = OfferToPDFConverter.class.getClassLoader()
            .getResource("offer-template/offer.html").toURI()
```
as a class variable and read the info into the doc like this (also a class variable) but in your case it would make more sense to do your call as you did already.

`    private static final OfferTemplate TEMPLATE = new OfferTemplate(Jsoup.parse(new File(OFFER_HTML_TEMPLATE_URI),"UTF-8"))`

Reply:
True but apparently loading a resource from a jar is only possible via an Inputstream which is then consumed. I wasn't able to get it to work with an URI :(